packages/map-template/src/components/QRCodeDialog/QRCodeDialog.jsx

```diff
@@ -0,0 +1,117 @@
+        if (directionsFrom && directionsTo) {
+            // The current URL when the user opens the QR code dialog
+            let currentUrl = window.location.origin;
+
+            // The interface for the existing URL search params
+            const currentParams = new URLSearchParams(window.location.search);
+
+            // The interface for the new URL search params
+            const newParams = new URLSearchParams();
+
+            /**
+             * Function that handles the presence of query parameters
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParams(queryParam) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                }
+            }
+
+            /**
+             * Function that handles the presence of query parameters and props
+             * and appends them to the newParams interface.
+             * 
+             * @param {string} queryParam 
+             * @param {string} prop 
+             */
+            function handleQueryParamsAndProps(queryParam, prop) {
+                if (currentParams.has(queryParam)) {
+                    const queryParameter = currentParams.get(queryParam);
+                    newParams.append(queryParam, queryParameter);
+                } else if (prop) {
+                    if (prop === primaryColorProp) {
+                        newParams.append(queryParam, primaryColorProp.replace("#", ""))
+                    } else {
+                        newParams.append(queryParam, prop)
+                    }
+                }
+            }
+
+            // Handle query parameters for the gmApiKey and mapboxAccessToken
+            handleQueryParams('gmApiKey');
+            handleQueryParams('mapboxAccessToken');
+
+            // Handle query parameters and props for apiKey, primaryColor, logo 
+            handleQueryParamsAndProps('apiKey', apiKeyProp);
+            handleQueryParamsAndProps('primaryColor', primaryColorProp);
+            handleQueryParamsAndProps('logo', logoProp);
+
+            // Get the string with all the final query parameters
+            const finalParams = newParams.toString()
+
+            // Construct the QR code URL
+            let QRCodeURL = `${currentUrl}/?${finalParams}&directionsFrom=${directionsFrom.id}&directionsTo=${directionsTo.id}`
+
+            const options = {
+                errorCorrectionLevel: 'L',
+                margin: 0,
+                width: '225'
+            };
+
+            QRCode.toDataURL(QRCodeURL, options)
+                .then((dataUrl) => {
+                    document.getElementById('qr').src = dataUrl;
+                }).catch(() => {
+                    console.log('error')
```

Comment:
Nah 😄

Reply:
Hah, I must have missed that. 😅 What should happen in the `catch`?

Reply:
Do you have any suggestion on how we should handle that?

Reply:
The question is if we should handle it at all. What can go wrong? Have you seen it failing?

Reply:
Hmm no actually. I will remove the `catch` code block.